pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -435,12 +433,20 @@ func (plugin *cniNetworkPlugin) buildCNIRuntimeConf(podName string, podNs string
 			// see: https://github.com/containernetworking/cni/blob/master/CONVENTIONS.md and
 			// https://github.com/containernetworking/plugins/blob/master/plugins/meta/bandwidth/README.md
 			// Rates are in bits per second, burst values are in bits.
-			bandwidthParam.IngressRate = int(ingress.Value())
-			bandwidthParam.IngressBurst = math.MaxInt32 // no limit
+			bandwidthParam.IngressRate = int(ingress.Rate.Value())
+			if ingress.Burst != nil {
+				bandwidthParam.IngressBurst = int(ingress.Burst.Value())
+			} else {
+				bandwidthParam.IngressBurst = math.MaxInt32
+			}
 		}
 		if egress != nil {
-			bandwidthParam.EgressRate = int(egress.Value())
-			bandwidthParam.EgressBurst = math.MaxInt32 // no limit
+			bandwidthParam.EgressRate = int(egress.Rate.Value())
+			if egress.Burst != nil {
+				bandwidthParam.EgressBurst = int(egress.Burst.Value())
```

Comment:
I'm not good with this type conversions, but my understanding is that [`int` may be 32 or 64 bits](https://stackoverflow.com/questions/21491488/what-is-the-difference-between-int-and-int64-in-go).
Should we cap it to `math.MaxInt32`?

Reply:
Hi aojea, thanks for you advice. I followed previous implementation on assigning int64 rate value to `cniBandwidthEntry` field of int, see deleted line 442 of this file. 

And I think you're right about `int`, see: https://golang.org/pkg/builtin/#int, it's platform specific. I also did some experiment on it, actually `int(int64)` will remain `int64`'s size and will not be truncated on 64-bit platform, but on 32-bit platform(like the playground) it will result an error: https://play.golang.org/p/c34Z2QkPfKY.

If supporting 32-bit platform is not a goal, and considering rate and burst are in bit, they could be quit large, I don't think capping it to `MaxInt32`(just around 2147 Mbit) is a good idea.

Reply:
thanks for the explanation, it makes sense

Reply:
looking at the CNI bandwidth plugin it seems that `burst/8 >= math.MaxUint32:` is not possible

https://github.com/containernetworking/plugins/blob/f5c3d1b1bab6ee72a32702c22ebb0f008580ef2d/plugins/meta/bandwidth/main.go#L113-L114